library/bignum.c

```diff
@@ -200,7 +200,13 @@ int mbedtls_mpi_copy( mbedtls_mpi *X, const mbedtls_mpi *Y )
 
     MBEDTLS_MPI_CHK( mbedtls_mpi_grow( X, i ) );
 
-    memset( X->p, 0, X->n * ciL );
+    if( X->n > i )
```

Comment:
```if( i > X->n )``` is checked inside ```mbedtls_mpi_grow( X, i )```. If we change here we have to repeat the condition here. Since, else can be ```i == X->n```.